src/main/java/net/sf/jabref/logic/integrity/ISSNChecker.java

```diff
@@ -0,0 +1,53 @@
+package net.sf.jabref.logic.integrity;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.logic.integrity.IntegrityCheck.Checker;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+
+
+public class ISSNChecker implements Checker {
+
+    private static final Pattern ISSN_PATTERN = Pattern.compile("^\\d{4}-\\d{3}[\\dxX]$");
+
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        if (!entry.hasField("issn")) {
+            return Collections.emptyList();
+        }
+
+        // Check that the ISSN is on the correct form
+        List<IntegrityMessage> result = new ArrayList<>();
+        String issn = entry.getField("issn").trim();
+        Matcher issn_matcher = ISSN_PATTERN.matcher(issn);
+        if (!issn_matcher.matches()) {
```

Comment:
Bah! Quick hack, so clearly didn't spend much time thinking about those
things... Will fix!

I found an ISSN where the checker claims to give an incorrect control
digit. It happens when there's a zero control (final) digit and it makes
sense that the current code doesn't work then. One more if-statement
needed...